### ISSUE TITLE
Batch Normalization for 5D tensors (NxDxHxWxC) to handle 3D data (e.g., medical images)

### ISSUE THREAD

**Issue by dibenedetto** [tensorflow/tfjs#5843]
**System information**
- TensorFlow.js version (you are using): 3.11.0
- Are you willing to contribute it (Yes/No): No

**Describe the feature and the current behavior/state.**

Batch Normalization (BN) in 5D tensors (NxDxHxWxC) is needed to handle models dealing with single- or multi-channel gridded 3D data, like 3D CT or MR medical images.

Currently BN is limited up to 4D tensors, and with 5D ones it fails with the following error in <a href="https://github.com/tensorflow/tfjs/blob/bec2196c8861470f929a3afee74eef1f10a2258c/tfjs-layers/src/layers/normalization.ts#L66">normalization.ts</a>:
`Error: batchNormalization is not implemented for array of rank 5 yet`

This is basically a re-open of the closed issue #2504, which said: 
> Ssupport for 5D tensors/3D ops has been added to several components recently (#767 #1035 #1985), so I am filing this hoping that someone with free time will consider adding 5D tensor support for BatchNorm too.

**Will this change the current api? How?**
This will **not** change the current API.

**Who will benefit with this feature?**
Anyone willing to use 3D tensors in their models (e.g., medical imaging).

**Any Other info.**
At first vision of the source code, to me it wouldn't take too long to implement, but I can't do it right now because I have problems in rebuilding the whole repo.

**Comment by pyu10055** [comment]
@dibenedetto Is this support aiming for training (layers API) or inference (Graph model execution)?

**Comment by dibenedetto** [comment]
> @dibenedetto do you need this support for training (layers API) or inference (Graph model execution)?

at this time for inference.

right now i bypassed the 5D BN by reshaping the tensor to 3D and then back to 5D, and this hack seems to work, but having a working 5D BN would avoid keeping track of tensor shapes in model definition.

**Comment by gaikwadrahul8** [comment]
Hi, @dibenedetto 

Apologize for the delayed response and we are re-visiting our older feature requests and checking whether those feature requests implemented or not as of now and it seems like we haven't implemented this feature yet if I'm not wrong so May I know are you still looking for this feature in TFJs ? 

https://github.com/tensorflow/tfjs/blob/bec2196c8861470f929a3afee74eef1f10a2258c/tfjs-layers/src/layers/normalization.ts#L44-L70

If someone wants to contribute for this feature then you're always welcome and please feel free to do and please refer these links [Ref-1](https://github.com/tensorflow/tfjs/blob/master/CONTRIBUTING.md), [Ref-2](https://github.com/tensorflow/tfjs/blob/master/CONTRIBUTING_MISSING_OP.md) . Thank you!
